bigflow/log.py

```diff
@@ -1,89 +1,136 @@
 import logging
+import typing
 import sys
+import uuid
+import traceback
+
+from textwrap import dedent
+
 from google.cloud import logging_v2
+from google.cloud.logging_v2.gapic.enums import LogSeverity
 from urllib.parse import quote_plus
 
+try:
+    from typing import TypedDict
+except ImportError:
+    TypedDict = dict
 
-def create_logging_client():
-    return logging_v2.LoggingServiceV2Client()
 
+class GCPLoggerHandler(logging.Handler):
 
-class GCPLoggerHandler(logging.StreamHandler):
-    def __init__(self, project_id, logger_name, workflow_id):
+    def __init__(self, project_id, log_name, labels):
         logging.StreamHandler.__init__(self)
-        self.client = create_logging_client()
+
+        self.client = self.create_logging_client()
         self.project_id = project_id
-        self.workflow_id = workflow_id
-        self.logger_name = logger_name
-
-    def emit(self, record):
-        if record.levelname == 'INFO':
-            self.info(record.getMessage())
-        elif record.levelname == 'WARNING':
-            self.warning(record.getMessage())
-        else:
-            self.error(record.getMessage())
-
-    def get_resource(self):
-        return {
-            "type": "global",
-            "labels": {
-                "project_id": f"{self.project_id}"
-            }
+        self.log_name = log_name
+        self.labels = labels
+
+        self._log_entry_prototype = logging_v2.types.LogEntry(
+            log_name=f"projects/{self.project_id}/logs/{self.log_name}",
+            labels=dict(labels),
+            resource={
+                "type": "global",
+                "labels": {
+                    "project_id": str(self.project_id),
+                },
+            },
+        )
+
+    def create_logging_client(self):
+        return logging_v2.LoggingServiceV2Client()
+
+    def emit(self, record: logging.LogRecord):
+        cl_log_level = record.levelname  # CloudLogging list of supported log levels is a superset of python logging level names
```

Comment:
I think we can remove this comment

Reply:
Hm, I'd like to keep it (at least for me it is not obvious that we don't need to perform sort of mapping between python & CL logging levels).